src/Elmish.WPF/BindingVmHelpers.fs

```diff
@@ -613,22 +614,24 @@ type Update
           this.Recursive(currentModel |> ValueOption.map b.Get, getCurrentModel >> b.Get, b.Get newModel, b.Dispatch, b.Binding)
 
 
-type Get(nameChain: string) =
+type Get<'t>(nameChain: string) =
 
-  member _.Base (model: 'model, binding: BaseVmBinding<'model, 'msg>) =
+  member _.Base (model: 'model, binding: BaseVmBinding<'model, 'msg, 't>) =
     match binding with
     | OneWay { OneWayData = d } -> d.Get model |> Ok
     | TwoWay b -> b.Get model |> Ok
     | OneWayToSource _ -> GetError.OneWayToSource |> Error
     | OneWaySeq { Values = vals } -> vals.GetCollection () |> Ok
-    | Cmd cmd -> cmd |> box |> Ok
-    | SubModel { Vm = vm } -> vm.Value |> ValueOption.toObj |> box |> Ok
+    | Cmd cmd -> cmd |> unbox |> Ok
+    | SubModel { Vm = vm } ->
+        vm.Value
+        |> ValueOption.toNull
+        |> Result.mapError GetError.ToNullError
```

Comment:
So if I don't deal with this case, it will constrain the `'t` at the top of the function to only work on nullable types. This allows `'t` to work on non-nullable types as well, while mapping the case of trying to unwrap a value of `ValueNone` when the type is non-nullable into an error case.

The core issue here is that `SubModel`'s vm property is fundamentally nullable, which is a problem for SubModels that could be non-nullable types.

The other bigger issue is that we need `'t` to be unconstrained because we are using the same `Get` functions for getting primitives as we are using to get sub models.

Reply:
Pin this discussion for a later PR (reduced scope of this PR)